### PR TITLE
Updates the timing callback to have consistent trainer.global_step: Amplify, ESM2, Geneformer, Evo2

### PR DESCRIPTION
**Pull request by jomitchellnv**
This is a follow-up on the MR that updates the timing callback from ESM2 https://github.com/NVIDIA/bionemo-framework/pull/914

This simply applies it to other sub-packages.

### Type of changes

- [X]  Bug fix (non-breaking change which fixes an issue)
- [ ]  New feature (non-breaking change which adds functionality)
- [ ]  Refactor
- [ ]  Documentation update
- [ ]  Other (please describe):

### CI Pipeline Configuration
Configure CI behavior by applying the relevant labels:

- [SKIP_CI](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#skip_ci) - Skip all continuous integration tests
- [INCLUDE_NOTEBOOKS_TESTS](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#include_notebooks_tests) - Execute notebook validation tests in pytest
- [INCLUDE_SLOW_TESTS](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#include_slow_tests) - Execute tests labelled as slow in pytest for extensive testing

> [!NOTE]
> By default, the notebooks validation tests are skipped unless explicitly enabled.

#### Authorizing CI Runs

We use [copy-pr-bot](https://docs.gha-runners.nvidia.com/apps/copy-pr-bot/#automation) to manage authorization of CI
runs on NVIDIA's compute resources.

* If a pull request is opened by a trusted user and contains only trusted changes, the pull request's code will
  automatically be copied to a pull-request/ prefixed branch in the source repository (e.g. pull-request/123)
* If a pull request is opened by an untrusted user or contains untrusted changes, an NVIDIA org member must leave an
  `/ok to test` comment on the pull request to trigger CI. This will need to be done for each new commit.

### Usage

```python
TODO: Add code snippet
```

### Pre-submit Checklist

 - [ ] I have tested these changes locally
 - [ ] I have updated the documentation accordingly
 - [ ] I have added/updated tests as needed
 - [ ] All existing tests pass successfully

### CODE REVIEW

**Comment by jstjohn** [comment]
Oh nice, this addresses #917 Please close that ticket once this merges.

**Review by jstjohn — APPROVED** [review]

**Comment by jstjohn** [comment]
I think you missed one:
```
sub-packages/bionemo-llm/src/bionemo/llm/train.py:from nemo.utils.exp_manager import TimingCallback
```

**Comment by dorotat-nv** [comment]
ok, but the tflops logger depends on the output of the `TimingCallback`, have you tested that the new callback is compatible? You can find the dependency here: https://github.com/NVIDIA/NeMo/blob/94ce722a80b2b108da1f033139afc8527402cfa1/nemo/lightning/pytorch/callbacks/flops_callback.py#L132

File: sub-packages/bionemo-amplify/src/bionemo/amplify/train_amplify.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
 from nemo.lightning.pytorch import callbacks as nl_callbacks
 from nemo.lightning.pytorch.callbacks.flops_callback import FLOPsMeasurementCallback
 from nemo.lightning.pytorch.optim import MegatronOptimizerModule
-from nemo.utils.exp_manager import TimingCallback
+from nemo.utils.exp_manager import DeltaTimingCallback
```
Comment (dorotat-nv) [inline sub-packages/bionemo-amplify/src/bionemo/amplify/train_amplify.py:30]: ok, but the tflops logger depends on the output of the TimingCallback, have you tested that the new callback is compatible? You can find the dependency here: https://github.com/NVIDIA/NeMo/blob/94ce722a80b2b108da1f033139afc8527402cfa1/nemo/lightning/pytorch/callbacks/flops_callback.py#L132

Can you confirm that this works? 

We have unit tests that check it, ie 
*  https://github.com/NVIDIA/bionemo-framework/blob/2efeba63f54428fca345f858f3374cd789f159b5/sub-packages/bionemo-esm2/tests/bionemo/esm2/scripts/test_train_esm2.py#L330
* https://github.com/NVIDIA/bionemo-framework/blob/2efeba63f54428fca345f858f3374cd789f159b5/sub-packages/bionemo-evo2/tests/bionemo/evo2/run/test_train.py#L150
 But they are marked as slow. Could you confirm if they test it correctly?

Comment (dorotat-nv) [reply sub-packages/bionemo-amplify/src/bionemo/amplify/train_amplify.py:30]: please rerun the testing pipeline with added labels
